doc/getting-started.md

```diff
@@ -0,0 +1,584 @@
+# Getting Started
+
+This section will serve as an introductory Mio tutorial. It assumes that
+you have familiarity with the [Rust](http://www.rust-lang.org/)
+programming language and the [Cargo](https://crates.io) tool. It will
+start from generating a new [Rust](http://www.rust-lang.org/) project
+using [Cargo](https://crates.io) up to writing a simple TCP echo server
+and client.
+
+Of course, you will need Rust installed. If you haven't already, get it
+here: [rust-lang.org](https://www.rust-lang.org).
+
+The complete echo server can be found
+[here](../examples/ping_pong/src/server.rs). A client example is
+[here](../examples/ping_pong/src/client.rs).
+
+> **Note:** As of the time of writing, Mio does not support Windows
+> (though support is currently in progress).
+
+Before you get started, setup an empty text file. Write down any
+thoughts, confusions, questions that you have while going through this
+guide. You only have one first impression, and I would like to capture
+this in order to improve this document.
+
+Post your notes as a comment [here](https://github.com/carllerche/mio/pull/222).
+
+## Setting up the project
+
+The first step is getting a new Cargo project setup. In a new
+directory, run the following:
+
+```not_rust
+cargo new pingpong --bin
+cd pingpong
+```
+
+Now, open the directory in your favorite text editor. You should see the
+following files:
+
+* src/main.rs
+* Cargo.toml
+
+If you are not already familiar with Cargo, you can learn more about it
+[here](http://doc.crates.io/).
+
+Open `Cargo.toml` and add a dependency on Mio by putting the following a
+the bottom of the file:
+
+```toml
+[dependencies]
+mio = "0.4.1"
+```
+
+Save the file, then compile and run the project using the following
+command:
+
+```not_rust
+cargo run
+```
+
+You will see some Cargo related output followed by `Hello, world!`. We
+haven't written any code yet and this is the default behavior of a
+freshly generated Cargo project.
+
+## Writing the Echo Server
+
+Let's start by writing a very simple server that accepts connections and
+does nothing with them. The connections will be accepted and shutdown
+immediately after.
+
+Here is the entire code, we'll step through it in a bit.
+
+```rust
+extern crate mio;
+
+use mio::tcp::*;
```

Comment:
I'd avoid glob imports both here and in example code, though it can make the final code a bit briefer, it makes it much harder to follow since you don't know where symbols came from.